File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.js
/**
 * @param {string} s
 * @return {number}
 */
var minOperations = function(s) {
    const minOps = (start) => {
        let ops = 0;
        for (let c of s) {
            ops += (c == start) ? 1 : 0;
            start ^= 1;
        }
        return ops;
    }
    return Math.min(minOps(0), minOps(1));
};

File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.java
class Solution {
    public int minOperations(String s) {
        return Math.min(minOps(s, '0'), minOps(s, '1'));
    }

    public int minOps(String s, char start) {
        int ops = 0;
        for (char c : s.toCharArray()) {
            ops += (c == start) ? 0 : 1;
            start = (char) (start ^ 1);
        }
        return ops;
    }
}
